var/spack/repos/builtin/packages/hdf5-vol-async/package.py

```diff
@@ -24,7 +24,8 @@ class Hdf5VolAsync(CMakePackage):
 
     depends_on("mpi")
     depends_on("argobots@main")
-    depends_on("hdf5@1.13: +mpi +threadsafe")
+    depends_on("hdf5@1.13.0:1.13.2 +mpi +threadsafe")
+    depends_on("hdf5@1.13.3: +mpi +threadsafe", when="@1.4:")
```

Comment:
This gives no possible versions for hdf5 when @1.4:
```suggestion
    depends_on("hdf5 +mpi +threadsafe")
    depends_on("hdf5@1.13.0:1.13.2", when="@:1.3")
    depends_on("hdf5@1.13.3:", when="@1.4:")
```